### PR TITLE
[LinalgExt] Add a decomposition pass to decompose AggregatedOpInterface Ops

### PR DESCRIPTION
**Pull request by Abhishek-Varma**
-- This commit creates a new pass `--iree-linalg-decompose` to decompose
   Linalg-like ops which have implemented AggregatedOpInterface.
-- Currently only linalg::Softmax implements it and hence this takes care
   of the same, but would slowly capture linalg_ext::attention as well as
   linalg_ext::winograd* ops.
   
Signed-off-by: Abhishek Varma <abhishek@nod-labs.com>

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/Passes.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@ void addCommonTargetExecutablePreprocessingPasses(OpPassManager &passManager) {
   nestedModulePM.addPass(createBubbleUpOrdinalOpsPass());
   nestedModulePM.addPass(createBufferizeCopyOnlyDispatchesPass());
   nestedModulePM.addNestedPass<func::FuncOp>(
-      IREE::LinalgExt::createDecomposeSoftmaxPass());
+      IREE::LinalgExt::createDecomposeLinalgLikeOpsPass());
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/Common/Passes.cpp]: I would expect that we are dropping this pass completely?

After dropping this, the tile and distrubute should pick up softmax, and the same pass added to LLVMCPU/Passes.cpp / LLVMGPU/Passes.cpp / SPIRV/Passes.cpp should take care of the decomposition.

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

File: compiler/src/iree/compiler/Codegen/Common/Passes.cpp (hunk 1 above)
Comment (Abhishek-Varma) [reply compiler/src/iree/compiler/Codegen/Common/Passes.cpp]: Aah! Good catch! I forgot to remove the pass from here while testing the softmax fusion construct.
Have checked locally after removing it that lit tests still pass - so pushed an updated revision to this PR. :)

File: llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/Passes/Passes.h [hunk 2]
```diff
@@ -180,6 +176,10 @@ tileAndDecomposeAttention(IREE::LinalgExt::AttentionOp attnOp,
 // linalg ops.
 std::unique_ptr<Pass> createTileAndDecomposeAttentionPass();
 
+// Creates a pass to decompose certain linalg-like ops into a sequence of linalg
+// ops.
+std::unique_ptr<Pass> createDecomposeLinalgLikeOpsPass();
```
Comment (MaheshRavishankar) [inline llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/Passes/Passes.h]: Nit: Please rename to `createDecomposeAggregateOpsPass`.

File: llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/Passes/Passes.td [hunk 3]
```diff
@@ -103,4 +95,12 @@ def TileAndDecomposeAttention :
   ];
 }
 
+def DecomposeLinalgLikeOps :
```
Comment (MaheshRavishankar) [inline llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/Passes/Passes.td]: Nit: rename to `DecomposeAggregateOps`.

**Review by MaheshRavishankar — COMMENTED** [review]
Looks good. Just a few nits.

**Comment by benvanik** [comment]
Closing stale PR.

**Pull request closed without merging**